app/views/protocols/protocolsio_import_create.js.erb

```diff
@@ -13,21 +13,15 @@ $('#pio_no_file_error_span').addClass('has-error').html('<span class="help-block
 <% @protocolsio_no_file = false %>
 <% else %>
   $('#modal-import-json-protocol').modal('hide');
-      <% if remotipart_submitted? %> <%# a workaround to a bug with remotipart, that caused alot of headache, courtesy of github.com/dhampik %>
-        $('#protocolsio-preview-modal-target').html(
-          "<%= j "#{render(:partial => 'protocols/import_export/import_json_protocol_preview_modal')}" %>"
-        );
-      <% else %>
-        $('#protocolsio-preview-modal-target').html(
-          "<%= j render(:partial => 'protocols/import_export/import_json_protocol_preview_modal') %>"
-        );
-      <% end %>
-      $('#modal-import-json-protocol-preview').modal('show');
-      $('.modal').on('hidden.bs.modal', function (e) {
-        if($('.modal').hasClass('in')) {
-          $('body').addClass('modal-open');
-        }
-      });
-
-
+  $('#protocolsio-preview-modal-target').html(
```

Comment:
I don't know, this is definitely not OK to have js code in view and handle this in that way. At least you can make an endpoint that will return this modal and then initialize it,...

Reply:
As far as i was aware this is normal for a js response, i made it this way (in a .js.erb file) because of this post below, that said it was expected behaviour. The view is never shown, it is just a script that executes all the code since they are ajax requests. I can change it, but then i have to know what my format.js request calls.
https://stackoverflow.com/questions/13545148/rails-respond-to-format-js-api?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa